### PR TITLE
fixed a newly introduced regression in softmax

### PR DESCRIPTION
**Pull request by ktarplee**
There is a regression in softmin in 0.4.1 that was not present in 0.4.0.  The behavior of softmin(x) should match softmax(-x) however instead it is implemented (in v0.4.1) as -softmax(x).  These are not the same.  The fix is trivial because the bug is due to operator precedence.

This is a major regression that broke my training.  I'm not sure how a unit test did not catch this.

```
x = torch.tensor([1, 2, 3.5, 4])
print(F.softmin(x, dim=0)) # this has the wrong output in 0.4.1 but correct in 0.4.0
print(F.softmax(-x, dim=0)) # this is what softmax should be
print(F.softmax(x, dim=0))
print(-F.softmax(x, dim=0)) # this is how softmax is implemented incorrectly
```
In 0.4.1 this produces
tensor([-0.0278, -0.0755, -0.3385, -0.5581])
tensor([0.6668, 0.2453, 0.0547, 0.0332])
tensor([0.0278, 0.0755, 0.3385, 0.5581])
tensor([-0.0278, -0.0755, -0.3385, -0.5581])

In 0.4.0 this produces the correct values
tensor([ 0.6668,  0.2453,  0.0547,  0.0332])
tensor([ 0.6668,  0.2453,  0.0547,  0.0332])
tensor([ 0.0278,  0.0755,  0.3385,  0.5581])
tensor([-0.0278, -0.0755, -0.3385, -0.5581])

### CODE REVIEW

**Comment by soumith** [comment]
this is bad, sorry about the regression. Would you be up for adding a test in `test_nn.py`?

**Review by soumith — APPROVED** [review]

**Comment by ktarplee** [comment]
I would be happy to add to test_nn.py however it looks like somewhat of a pain to build and test pytorch from source.

**Comment by soumith** [comment]
@ktarplee no worries, we'll add it and push this PR through.

**Comment by SsnL** [comment]
Thanks for the fix. Can you also instead submit this PR to the `master` branch?

**Comment by soumith** [comment]
@SsnL i've changed it to be on master.

**Review by soumith — APPROVED** [review]

**Review by soumith — APPROVED** [review]
